tests/bgp/test_bgp_speaker.py

```diff
@@ -199,3 +224,68 @@ def test_bgp_speaker_announce_routes(common_setup_teardown, testbed, duthost, pt
                         "testbed_mtu": mtu },
                 log_file="/tmp/bgp_speaker_test.FibTest.log",
                 socket_recv_size=16384)
+
+
+@pytest.mark.parametrize("ipv4, ipv6, mtu", [pytest.param(False, True, 1514)])
+def test_bgp_speaker_announce_routes_v6(common_setup_teardown, testbed, duthost, ptfhost, ipv4, ipv6, mtu, collect_techsupport):
+    """Setup bgp speaker on T0 topology and verify routes advertised by bgp speaker is received by T0 TOR
+
+    """
+    ptfip, mg_facts, interface_facts, vlan_ips, vlan_if_name, nexthops_ipv6, speaker_ips, port_num, http_ready = common_setup_teardown
+    assert http_ready
+
+    logging.info("announce route")
+    peer_range = mg_facts['minigraph_bgp_peers_with_range'][0]['ip_range'][0]
+    lo_addr = mg_facts['minigraph_lo_interfaces'][0]['addr']
+
+    logging.info("Announce ipv6 prefixes over ipv4 bgp sessions")
+    prefix = 'fc00:10::/64'
+    announce_route(ptfip, lo_addr, prefix, nexthops_ipv6[1].ip, port_num[0])
+    announce_route(ptfip, lo_addr, prefix, nexthops_ipv6[2].ip, port_num[1])
+    announce_route(ptfip, lo_addr, peer_range, vlan_ips[0].ip, port_num[2])
+
+    logging.info("Wait some time to make sure routes announced to dynamic bgp neighbors")
+    time.sleep(30)
+
+    # The ping here is workaround for known issue:
+    #     https://github.com/Azure/SONiC/issues/387 Pre-ARP support for static route config
+    # When there is no arp entry for next hop, routes learnt from exabgp will not be set down to ASIC
+    # Traffic to prefix 10.10.10.0 will be routed to vEOS VMs via default gateway.
+    duthost.shell("ping %s -c 3" % vlan_ips[1].ip)
+    duthost.shell("ping %s -c 3" % vlan_ips[2].ip)
+    time.sleep(5)
+
+    logging.info("Verify nexthops and nexthop interfaces for accepted prefixes of the dynamic neighbors")
```

Comment:
v4 check the bgp accepted prefix is correct. can we add the same check for v6 prefix? 
https://github.com/Azure/sonic-mgmt/pull/1944/files#diff-c19677d9f3957349e1b3f429f7ee77e0L173

Reply:
Currently we got incorrect number of prefixes from bgpfacts. There are some issues in the module. I have to fix them first.